PWGJE/EMCALJetTasks/AliEmcalJetTask.cxx

```diff
@@ -1085,28 +1085,33 @@ void AliEmcalJetTask::SetTrackingEfficiencyFromYAML(std::string period , std::st
   std::vector <Double_t> trackingUncertainty;
   bool res = fYAMLConfig.GetProperty("ptBinning", ptBinning, false);
   Int_t nPtBins = ptBinning.size()-1;
-  double* aptBinning = &ptBinning[0];
-
-  TString sPeriod(period);
+  double* aptBinning = ptBinning.data();
+
+  TObjArray *sPeriodCentArray = TString(periodCent).Tokenize("_");
+  TString sPeriod = sPeriodCentArray->At(0)->GetName();
+  TString sCent = "";
+  Bool_t isCentrality = kFALSE;
+  if(sPeriodCentArray->GetEntries()==3) {
+    sCent = Form("%s_%s",sPeriodCentArray->At(1)->GetName(),sPeriodCentArray->At(2)->GetName());
+    isCentrality = kTRUE;
+  }
+  std::string period = std::string(sPeriod.Data());
+  std::string cent = std::string(sCent.Data());
 
-  if(sPeriod=="LHC18q_010")       res = fYAMLConfig.GetProperty({"leadLead","LHC18q","central"},trackingUncertainty, false);
-  else if(sPeriod=="LHC18q_3050") res = fYAMLConfig.GetProperty({"leadLead","LHC18q","semiCentral"},trackingUncertainty, false);
-  else if(sPeriod=="LHC18r_010")  res = fYAMLConfig.GetProperty({"leadLead","LHC18r","central"},trackingUncertainty, false);
-  else if(sPeriod=="LHC18r_3050") res = fYAMLConfig.GetProperty({"leadLead","LHC18r","semiCentral"},trackingUncertainty, false);
-  else {
-    AliFatal("Setting tracking efficiency period but the period selected is not implemented at present"); 
+  if(isCentrality) res = fYAMLConfig.GetProperty({period,cent},trackingUncertainty, false);
+  else             res = fYAMLConfig.GetProperty(period,trackingUncertainty, false);
+  if(!res){
+    AliFatal(Form("Setting tracking efficiency period but the period %s and centrality %s selected is not implemented at present",period.c_str(),cent.c_str())); 
   }
 
   fTrackEfficiencyHistogram = new TH1D("fTrackEfficiencyHistogram","h",nPtBins,aptBinning);
 
-  AliInfoStream() << "Setting pT-dependent uncertainty" << "\n";
   for(Int_t i=0;i<nPtBins;i++) {
     fTrackEfficiencyHistogram->SetBinContent(i+1,trackingUncertainty.at(i));
-    AliInfoStream() << "pT " << ptBinning.at(i) << " - " << ptBinning.at(i+1) << " track uncertainty: "<< trackingUncertainty.at(i) << "\n";
+    AliDebug(2,Form("pT %f - %f \t track uncertainty: %f", ptBinning.at(i), ptBinning.at(i+1), trackingUncertainty.at(i)));
```

Comment:
Same here for `Form`. You can use `AliDebugStream(level)`, so being concrete, you could use:

```c++
AliDebugStream(2) << "pT " << ptBinning.at(i) << " - " << ptBinning.at(i+1) << " track uncertainty: "<< trackingUncertainty.at(i) << "\n";
```